### PR TITLE
Add dedicated Flask test coverage for search, workspaces, and export APIs (#101)

### LINKED ISSUE #101: Three API endpoints: add dedicated test coverage
#### Problem

The baseline eval states: "All 14 HTTP route handlers — the primary user-facing interface — have no direct tests. The export route, PDF generation, workspace listing, workspace detail, composers, and logs endpoints are all unverified." While the project has ~80 test cases across 10 test files covering utility functions and CLI logic, no test exercises the Flask route handlers through the test client. The eval's "test-coverage" cluster finding specifically flags three API endpoints that lack any coverage, leaving the primary user-facing contract completely unverified against regressions.

#### Acceptance Criteria

- [ ] At least three API endpoints have dedicated test files/functions using Flask's test client
- [ ] Priority endpoints to cover: `/api/search`, `/api/workspaces`, and `/api/export` (or the three most complex handlers)
- [ ] Each endpoint test covers: successful response (200), error response (400/404/500), and edge case (empty data, missing params)
- [ ] Tests use fixtures or mocks for SQLite database access (do not require real Cursor data)
- [ ] Tests verify response JSON structure matches the expected contract (key names, types)
- [ ] Tests pass in CI (if applicable)
- [ ] PR approved by at least 1 reviewer

#### Implementation Notes

Use Flask's built-in test client via `app.test_client()` from the `create_app()` factory. Create `tests/test_api_search.py`, `tests/test_api_workspaces.py`, and `tests/test_api_export.py`. For database mocking, create a temporary SQLite database with known test data mimicking Cursor's schema (`ItemTable` with `key`/`value` columns, `cursorDiskKV` patterns). Reference `tests/` for existing fixture patterns. The search endpoint is the highest priority given its 450-line complexity and three data sources. For the workspace endpoint, verify that the response includes the expected workspace metadata structure. For export, verify that the correct content-type headers and file format are returned. Note: if item #5 (search decomposition) lands first, the extracted helper functions will be easier to test in isolation — coordinate ordering with Brad.

#### References

- Eval finding: Test 30, cluster "test-coverage"
- Related files: `api/search.py`, `api/workspaces.py`, `api/export_api.py`, `app.py` (`create_app` factory), `tests/`, `requirements.txt`

### PR DESCRIPTION
**Pull request by bradjin8**
## Summary

- Split the monolithic `tests/test_api_endpoints.py` into three dedicated route test modules: `test_api_search.py`, `test_api_workspaces.py`, and `test_api_export.py`
- Add missing coverage for `POST /api/export` and `GET /api/export/state`, including zip response headers, export state persistence, and `since=last` behavior
- Each endpoint suite exercises happy path (200), error paths (400/404/500), and edge cases (empty data, missing params, exclusion rules) using the existing temp SQLite fixtures — no real Cursor data required
- Add shared `client_with_rules()` helper in `conftest.py` and update CI to run the new test files

Closes #101.

## Test plan

- [x] `python -m pytest tests/test_api_search.py tests/test_api_workspaces.py tests/test_api_export.py -v` (33 passed)
- [ ] CI unittest + pytest matrix passes on push

## Summary by CodeRabbit

* **Tests**
  * Reorganized API test suite into dedicated modules for export, search, and workspace endpoints; added broad coverage and new helpers.
* **Bug Fixes**
  * Export state handling: API now treats malformed saved state as empty, improving stability.
* **Chores**
  * Updated CI workflow to run the reorganized test subsets and clarified typehint guidance in typecheck job.

### CODE REVIEW

File: tests/test_api_workspaces.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    def test_global_returns_other_chats(self, client):
+        response = client.get("/api/workspaces/global")
+        assert response.status_code == 200
+        body = response.get_json()
+        assert body["id"] == "global"
+        assert body["name"] == "Other chats"
+
+
+class TestGetWorkspaceTabs:
```
Comment (timon0305) [inline tests/test_api_workspaces.py:72]: The split was supposed to keep all the old tests. This one for "global workspace + summary mode" is gone. Please add it back to TestGetWorkspaceTabs. Everything else carried over fine, so this looks accidental.

File: tests/conftest.py [hunk 2]
```diff
@@ -141,6 +142,19 @@ def client(workspace_storage: str):
     return app.test_client()
 
 
+def client_with_rules(rule_lines: list[str]) -> FlaskClient:
```
Comment (timon0305) [inline tests/conftest.py]: Importing helpers from conftest.py is discouraged (pytest treats it specially). Works, but move it to a normal file like tests/_helpers.py — same pattern you already use with tests/_fixture_ids.py. Minor.

File: tests/test_api_export.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    def test_since_last_after_export_returns_404_when_nothing_new(
+        self, client, export_state_dir
+    ):
+        first = _post_export(client, {"since": "all"})
+        assert first.status_code == 200
+
+        second = _post_export(client, {"since": "last"})
+        assert second.status_code == 404
+        body = second.get_json()
+        assert body.get("error") == "No conversations to export since last export"
```
Comment (timon0305) [inline tests/test_api_export.py:145]: This only passes because the seeded chat's timestamp is in the past. Fine now, but add a one-line comment saying so — otherwise changing fixture dates later would break this test in a confusing way.

**Review by timon0305 — APPROVED** [review]

File: tests/test_api_export.py (hunk 3 above)
Comment (bradjin8) [reply tests/test_api_export.py:145]: test_since_last_after_export_returns_404_when_nothing_new noting it depends on the seeded composer's May 2024 lastUpdatedAt being older than the export state written by the first call.

File: tests/test_api_workspaces.py (hunk 1 above)
Comment (bradjin8) [reply tests/test_api_workspaces.py:72]: Restored test_summary_query_global_workspace in TestGetWorkspaceTabs — verifies GET /api/workspaces/global/tabs?summary=1 returns 200 with a tabs list.

**Review by wpak-ai — APPROVED** [review]
